src/cpp/rtps/transport/shared_mem/SharedMemManager.hpp

```diff
@@ -612,10 +855,12 @@ class SharedMemManager
     {
         // Every buffer allocated implies two internal allocations, node and payload.
         // Every internal allocation consumes 'per_allocation_extra_size_' bytes
-        uint32_t allocation_extra_size = sizeof(SegmentNode) + per_allocation_extra_size_ +
-                max_allocations * ((sizeof(BufferNode) + per_allocation_extra_size_) + per_allocation_extra_size_);
+        uint32_t allocation_extra_size = 
+            sizeof(SegmentNode) + per_allocation_extra_size_ +
+            (max_allocations * sizeof(BufferNode)) + per_allocation_extra_size_ +
```

Comment:
I don't know if this should be the following (please explain):
```suggestion
            (max_allocations * (sizeof(BufferNode) + per_allocation_extra_size_) +
```

Reply:
The allocation algorithm used by boost consumes some extra memory by each alocation you do. That's because the internal tree used to store every buffer the allocator gives to you, is also in the shared memory segment, so not all the reserved size is free for your use. In the segment construction a test is perform to estimate how much extra memory is used per allocation (per_allocation_extra_size). So a segment is over-sized to be able to hold:
1) The segment node allocation.
2) The BufferNode pool allocation.
3) at least *max_allocations* number of user buffers

Reply:
I would update the comments, then